### PR TITLE
Fix pkey update issue

### PR DESCRIPTION
**Pull request by Michael-Tieying-Zhang**
This PR fixed the bug #384

### CODE REVIEW

**Comment by Michael-Tieying-Zhang** [comment]
I'll add something else. Pls merge it later when I am done.

File: src/executor/update_executor.cpp [hunk 1]
```diff
@@ -54,6 +54,73 @@ bool UpdateExecutor::DInit() {
   return true;
 }
 
+bool UpdateExecutor::PerformUpdatePrimaryKey(bool is_owner, oid_t tile_group_id,
+                                             oid_t physical_tuple_id,
+                                             ItemPointer &old_location,
+                                             storage::TileGroup *&tile_group) {
```
Comment (jessesleeping) [inline src/executor/update_executor.cpp]: We can pass tile_group by value here.

File: test/sql/update_primary_index_sql_test.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  EXPECT_EQ(result[6].second[0], '3');
+
+  // Perform normal update
+  SQLTestsUtil::ExecuteSQLQuery("UPDATE test SET b=2000 WHERE c=200", result,
+                                tuple_descriptor, rows_affected, error_message);
+
+  // test update result
+  SQLTestsUtil::ExecuteSQLQuery("SELECT * FROM test WHERE b=2000", result,
+                                tuple_descriptor, rows_affected, error_message);
+  // Check the return value
```
Comment (jessesleeping) [inline test/sql/update_primary_index_sql_test.cpp]: Add another query to test if we can actually update the primary key.

**Comment by Michael-Tieying-Zhang** [comment]
Done.

**Pull request merged**